File: contract/assembly/models/Review.ts
import { AccountID, getCurrentDate, getID } from "../utils";

class Review {
    id: string;
    creator: AccountID;
    text: string;
    rating: i32;
    recipeID: string;
    createdAt: string;

    constructor(creator: string, text: string, raiting: i32, recipeID: string) {
        this.id = getID();
        this.creator = creator;
        this.text = text;
        this.rating = raiting;
        this.recipeID = recipeID;
        this.createdAt = getCurrentDate();
    }
}

export default Review;

File: contracts/rust-contract/src/structs/Review.rs
use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
use near_sdk::serde::{Deserialize, Serialize};
use near_sdk::{AccountId};

#[derive(Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
#[serde(crate = "near_sdk::serde")]
pub struct Review {
    pub id: String,
    pub creator: AccountId,
    pub text: String,
    pub rating: f64,
    pub recipe_id: i128,
    pub created_at: String,
}